### ISSUE TITLE
Question about input to ASU

### ISSUE THREAD

**Issue by liamkl** [CMACH508/2021-WangZhicheng#6]
Hi team, 

I'm trying to understand your code. In this line below it looks like you are feeding a 4D tensor of shape [batch, feature, stocks, length] to a Conv1D layer:

https://github.com/CMACH508/DeepTrader/blob/4b00b2a88360a79d7739599dbe8fad29f7061ba3/src/model/ASU.py#L159

Am I misunderstanding something? Otherwise, wouldn't this raise an error? as Conv1D is expecting a 3D tensor of shape [batch, num_channels, length]. I appreciate any insight you can give. Thanks!

**Comment by YorkLouis** [comment]
I have encountered a similar issue.

**Comment by choasLC** [comment]
not just that
I also confuse that why they claimed the proposed market score unit can not be used in CSI 100, since it still can provides a risk signal to the agent; even though, Chinese stocks do not allow "short"

**Comment by SteamerLee** [comment]
That would be reasonable and make sense if changing all torch.nn.Conv1d() to torch.nn.Conv2d() since Conv1d cannot be applied to 4D tensor here.

**Comment by fxvvv** [comment]
> That would be reasonable and make sense if changing all torch.nn.Conv1d() to torch.nn.Conv2d() since Conv1d cannot be applied to 4D tensor here.

Can changing all torch.nn.Conv1d() to torch.nn.Conv2d() successfully produce results? Have you tried this approach? Thank you

**Comment by SteamerLee** [comment]
> > That would be reasonable and make sense if changing all torch.nn.Conv1d() to torch.nn.Conv2d() since Conv1d cannot be applied to 4D tensor here.
> 
> Can changing all torch.nn.Conv1d() to torch.nn.Conv2d() successfully produce results? Have you tried this approach? Thank you

Yes, I tried it, and it works.

**Comment by fxvvv** [comment]
> > > That would be reasonable and make sense if changing all torch.nn.Conv1d() to torch.nn.Conv2d() since Conv1d cannot be applied to 4D tensor here.
> > 
> > 
> > Can changing all torch.nn.Conv1d() to torch.nn.Conv2d() successfully produce results? Have you tried this approach? Thank you
> 
> Yes, I tried it, and it works.

Oh, thank you, i tried this method and it works. But then i meet the issue " loss has nan ", have you ever encountered the same problem？

**Comment by SteamerLee** [comment]
> > > > That would be reasonable and make sense if changing all torch.nn.Conv1d() to torch.nn.Conv2d() since Conv1d cannot be applied to 4D tensor here.
> > > 
> > > 
> > > Can changing all torch.nn.Conv1d() to torch.nn.Conv2d() successfully produce results? Have you tried this approach? Thank you
> > 
> > 
> > Yes, I tried it, and it works.
> 
> Oh, thank you, i tried this method and it works. But then i meet the issue " loss has nan ", have you ever encountered the same problem？

I got the normal losses before. But expect for changing the above Conv2d(), I may change other places for my tasks. So there may need other customizations when using in different tasks. Thank you!
